src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala

```diff
@@ -625,11 +633,23 @@ $DOCKERFILE_SIGIL
         }
         val destination = new File(imageDirectory, relativeDestination)
         if (source.exists) {
+          // The IO object's methods do not preserve executable bits, so we have to manually set
+          // these ourself.
           if (source.isDirectory) {
             IO.createDirectory(destination)
-            IO.copyDirectory(source, destination)
+            val toCopy = PathFinder(source).***.pair(Path.rebase(source, destination))
+            IO.copy(toCopy)
+            toCopy.foreach {
+              case (source, destination) =>
+                if (source.canExecute) {
+                  destination.setExecutable(true)
+                }
```

Comment:
I think the `copy` line beforehand will replace the target, so this might work regardless - but this is cleaner. Updated!